runtime/onert/core/include/backend/basic/BackendContextHelpers.h

```diff
@@ -219,25 +243,44 @@ template <typename T_BackendContext> ITensorRegistry *genTensors(T_BackendContex
 inline void initConsts(const ir::Operands &operands,
                        const util::Set<ir::OperandIndex> &external_operands,
                        ITensorRegistry *tensor_registry,
-                       const ir::OperandIndexMap<ir::OperandIndex> & /*shared_memory_operands_map*/)
+                       const ir::OperandIndexMap<ir::OperandIndex> &shared_memory_operands_map)
 {
   operands.iterate([&](const ir::OperandIndex &ind, const ir::Operand &operand) {
-    if (external_operands.contains(ind) || !operand.isConstant())
+    const bool has_const_shared_memory =
+      shared_memory_operands_map.find(ind) != std::end(shared_memory_operands_map) &&
+      operands.at(shared_memory_operands_map.at(ind)).isConstant();
+    const bool can_be_initialized_as_const = operand.isConstant() || has_const_shared_memory;
+    if (external_operands.contains(ind) || !can_be_initialized_as_const)
       return;
 
     auto tensor = tensor_registry->getNativeITensor(ind);
     assert(tensor != nullptr);
 
     VERBOSE(FillOperandData) << "Fill data for " << ind << std::endl;
 
-    auto data = operand.shareData();
-    assert(data && data->base());
-    ExternalTensor *ext_tensor = dynamic_cast<ExternalTensor *>(tensor);
-
-    if (ext_tensor == nullptr)
-      throw std::runtime_error{"This tensor is not external tensor"};
-
-    ext_tensor->setData(data);
+    if (has_const_shared_memory)
+    {
+      const auto &source_operand_ind = operands.at(shared_memory_operands_map.at(ind));
+      auto memory_source_data = source_operand_ind.shareData();
+      assert(memory_source_data && memory_source_data->base());
+      auto shared_mem_tensor = dynamic_cast<Tensor *>(tensor);
+      if (nullptr == shared_mem_tensor)
+      {
+        throw std::runtime_error{"Incorrect type of tensor to support sharing memory"};
+      }
+      shared_mem_tensor->setBuffer(const_cast<uint8_t *>(memory_source_data->base()));
+    }
+    else
```

Comment:
I think this `else` is dangerous in maintenance perspective because this function returns by in the other condition on the line 253, 254. Could you consider it?

Reply:
I am not sure if I fully understand. Note that in lines 253, 254 is used `can_be_initialized_as_const` and this `else` if based on `has_const_shared_memory`. But I see that the code can be unreadable. I've done some refactor + added some comments ;)

Reply:
This `else` statement is affected by changes made in lines 253 and 254. It doesn't seem ideal to have it dependent on conditions in another conditional block.

Reply:
How about following the comment https://github.com/Samsung/ONE/pull/14384#discussion_r1879732630 to resolve the problem I mentioned?

Reply:
@ragmani good point - I've used version proposed in https://github.com/Samsung/ONE/pull/14384#discussion_r1879732630